ush/create_gsi_info.sh

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env bash
+# Generic script to create convinfo, ozinfo, or satinfo for a given date.
+# Usage:
+#   ./create_gsi_info.sh <type> <date> [use2mobs]
+#   <type>: conv, oz, or sat
+#   <date>: date string to match
+#   [use2mobs]: (optional, only for conv) YES or NO
+
+set -eu
+# Inherit set -e inside of the get_usedate function below
+shopt -s inherit_errexit
+type_in=${1:-}
+date_in=${2:-}
+use2mobs=${3:-NO}
+
+if [[ -z "${type_in}" || -z "${date_in}" ]]; then
+  echo "Usage: ${0} <type> <date> [use2mobs]"
+  echo "  <type>: conv, oz, or sat"
+  echo "  <date>: date string to match"
+  echo "  [use2mobs]: (optional, only for conv, whether to use 2m observations) YES or NO"
+  exit 1
+fi
+
+# Function to get the most recent data available for the target obs.
+# If an empty string is returned, this represents an error.
+get_usedate() {
+  usedate=""
+  # Loop over files matching date pattern.
+  for datex in [1-2][0-9][0-9][0-9]*; do
+    # Skip for loop if there are no matches.
+    if [[ ! -e "${datex}" ]]; then
+      continue
+    fi
+
+    if [[ ${date_in} -ge ${datex} ]]; then
+      usedate=${datex}
+    fi
+  done
+
+  echo "${usedate}"
+}
+
+# Get the starting directory
+starting_dir="${DATA:-${PWD}}"
```

Comment:
Something to consider, I think it would be cleaner/more explicit if the script took in as an argument where it was writing out files to rather than it being implicitly $DATA. Just my $0.02